quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/PropertyMappingInterceptor.java

```diff
@@ -64,47 +66,64 @@ public static void enable() {
+
+        final Set<PropertyMapper<?>> allMappers = new HashSet<>(PropertyMappers.getMappers());
+
+        //TODO: this is still not a complete list - things like quarkus.log.console.enabled
+        // come from kc.log - but via a map from, not to.
+        // so we'd need additional logic like the getWildcardMappedFrom case for that
+
+        boolean filterRuntime = isRebuild() || Environment.isRebuildCheck();
+
+        var baseStream = StreamSupport.stream(iterable.spliterator(), false).flatMap(name -> {
+            PropertyMapper<?> mapper = PropertyMappers.getMapper(name);
+
+            if (mapper == null) {
+                return Stream.of(name);
             }
-        }
+            if (filterRuntime && mapper.getCategory() == OptionCategory.CONFIG) {
+                return Stream.of(); // advertising the keystore type causes the keystore to be used early
+            }
+            allMappers.remove(mapper);
+
+            if (!mapper.hasWildcard()) {
+                var wildCard = PropertyMappers.getWildcardMappedFrom(mapper.getOption());
+                if (wildCard != null) {
+                    ConfigValue value = context.proceed(name);
+                    if (value != null && value.getValue() != null) {
+                        return Stream.concat(Stream.of(name), wildCard.getToFromWildcardTransformer(value.getValue()));
+                    }
+                }
+            }
+
+            mapper = mapper.forKey(name);
+
+            // there is a corner case here -1 for the reload period has no 'to' value.
+            // if that becomes an issue we could use more metadata to perform a full mapping
+            return toDistinctStream(name, mapper.getTo());
+        });
+
+        var defaultStream = allMappers.stream()
+                .filter(m -> !m.getDefaultValue().isEmpty() && !m.hasWildcard()
+                        && m.getCategory() != OptionCategory.CONFIG) // advertising the keystore type causes the keystore to be used early
+                .flatMap(m -> toDistinctStream(m.getTo()));
+
+        return IteratorUtils.chainedIterator(baseStream.iterator(), defaultStream.iterator());
+    }
 
-        // this could be optimized by filtering the wildcard names in the stream above
-        return filterRuntime(IteratorUtils.chainedIterator(mappedWildcardNames.iterator(), context.iterateNames()));
+    private static Stream<String> toDistinctStream(String... values) {
+        return Stream.of(values).filter(Objects::nonNull).distinct();
     }
 
     @Override
     public ConfigValue getValue(ConfigSourceInterceptorContext context, String name) {
         if (Boolean.TRUE.equals(disable.get())) {
             return context.proceed(name);
         }
-        return PropertyMappers.getValue(context, name);
+        Function<String, ConfigValue> resolver = (n) -> PropertyMappers.getValue(context, n);
+        return NestedPropertyMappingInterceptor.resolve(resolver, resolver, name);
```

Comment:
Can you please elaborate why we need to manually call the interceptor here? Won't it get called implicitly.

Adding a comment would be great.

Reply:
Ok added something.

Reply:
If there is no recursion concern, why not to call the Property Mapper directly? What is the reason to call the nested interceptor?

Reply:
The comment is trying, but may be failing, to explain that we are doing this to track that we're resolving "name".  Otherwise if we proceed the nested interceptor could immediately restart the chain for "name".

Reply:
Right. So maybe we could state explicitly that it is to prevent immediate context restart?

Reply:
Ok made a further refinement and comment. In trying to reuse the logic, the intent of the method wasn't as clear as it should be. See if it reads better now.

Reply:
Should be better now, just a little more encapsulated.